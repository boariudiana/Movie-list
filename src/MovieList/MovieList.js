import React from "react";
import { Button, Paper, Grid, Typography } from "@material-ui/core";
import AddIcon from "@material-ui/icons/Add";
import styles from "./MovieList.module.css";
import {useMediaQuery} from '@material-ui/core'
import theme from '../../src/theme'

const MovieList = (props) => {

  const matches = useMediaQuery(theme.breakpoints.up("md"));


  return (
    <div className={styles.box}>
      <ul className={styles.list}>
        {props.movies.map((movie) => (
          <Paper>
            <li className={styles.listItem} key={movie.id}>
              <Grid container>
                <Grid item md={6} xs={6} sm = {6}>
                  <img
                    src={`https://image.tmdb.org/t/p/w154${movie.poster_path}`}
                    alt={movie.title}
                  />
                </Grid>
                <Grid container item md={6} xs={6} sm = {6} wrap="nowrap" direction ={matches ? "row": "column"} justify= "flex-start" alignItems = "flex-start">
                  <Grid item md={12} sm = {12} xs={12} >
                    <Typography
                      variant="h6"
                      align="left"
                      color="primary"
                    >
                      {movie.title}
                    </Typography>
                    <Typography 
                       variant = "body1"
                       align= "left"
                       color = "primary"
                    >
                      ({movie.release_date})
                      </Typography>
                      {/*<Typography
                        noWrap
                         variant = "body1"
                      align= "left">{movie.overview}</Typography>*/}
                  </Grid>
                  <Grid item md={12} xs={12} sm = {12}>
                    <Button
                      className={styles.addMovie}
                      color="secondary"
                      onClick={(e) => {
                        e.preventDefault();
                        props.onMovieAdd(movie);
                      }}
                    >
                      <AddIcon /> Add movie
                    </Button>
                  </Grid>
                </Grid>
              </Grid>
            </li>
          </Paper>
        ))}
      </ul>
    </div>
  );
};
export default MovieList;
