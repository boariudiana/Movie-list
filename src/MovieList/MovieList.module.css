.box {
    display: flex;
    justify-content: center;
  }
  
  .list {
    list-style: none;
    align-items: center;
    width: 60%;
  }
  
  .listItem {
    width:100%;
    height: 200px;
    margin-bottom: 16px;
  }
  .listItem img {
    max-height: 200px;
    border-radius: 3px;
  }
  
  
  .listItem .addMovie {
    display: none;
  }
  
  .listItem:hover .addMovie {
    display: inline-flex;
  }
  
  .listItem:hover {
      background-color: #F5F5F5;
  }
   @media(max-width : 450px){
     .list {
       width: 100%;
     }
   }

   @media (min-width: 800px){
    .listItem {
      height: 150px;
    }
    .listItem img {
      max-height: 150px;
    }
   }

   .overview{
     display: none;
   }
   
   @media(min-width: 960px){
      .overview{
        display: block;
      }
   }
  
