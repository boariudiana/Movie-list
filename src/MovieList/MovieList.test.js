import React from "react";
import  { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import MovieList from './MovieList';

configure({ adapter : new Adapter() })

const mockMovies = [
    {
        id : 12,
    } 
];


describe('MovieList Component' , () => {
    it("renders MovieList", () => {
      
        const wrapper = shallow( <MovieList  movies = {mockMovies}/> );
        expect(wrapper).toMatchSnapshot();
    })

    it("returns an empty array when there is no data to map though", () => {
      
        const wrapper = shallow( < MovieList  /> );
        expect(wrapper).toMatchSnapshot();
    })

    it( 'does not break with an empty array',  () => {
        const wrapper = shallow ( < MovieList  movies = {[]} /> );

        expect(wrapper.find('li')).toHaveLength(0);
    } )

    it( 'does not break without movies',  () => {
        const wrapper = shallow ( < MovieList /> );

        expect(wrapper.find('li')).toHaveLength(0);
    } )

    it( 'renders search results when movies changes', () => {
        const wrapper = shallow ( < MovieList 
                                          movies ={[]} /> );
        wrapper.setProps({
            movies: [{
                id : 12,
                title : "title"
            }]
        });
        expect(wrapper.find('img').prop("alt")).toEqual("title")
    } )

})

