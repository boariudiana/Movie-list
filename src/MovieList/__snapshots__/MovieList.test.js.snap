// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList Component renders MovieList 1`] = `ShallowWrapper {}`;

exports[`MovieList Component returns an empty array when there is no data to map though 1`] = `ShallowWrapper {}`;
